src/primitives/rectangle/mod.rs

```diff
@@ -239,6 +243,162 @@ impl Rectangle {
         // No overlap present
         Rectangle::zero()
     }
+
+    /// Resizes the rectangle relative to an anchor point.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use embedded_graphics::{prelude::*, primitives::rectangle::{Rectangle, AnchorPoint}};
+    ///
+    /// let mut rect = Rectangle::new(Point::new(20, 20), Size::new(10, 20));
+    /// rect.resize(Size::new(20, 10), AnchorPoint::Center);
+    ///
+    /// assert_eq!(rect, Rectangle::new(Point::new(15, 25), Size::new(20, 10)));
+    /// ```
+    pub fn resize(&mut self, size: Size, anchor_point: AnchorPoint) {
```

Comment:
I think this is related to #211 and we need to find a good consistent solution for all transformations of the primitives. But I'm not exactly sure what that is at the moment.

Was the `self` in you suggestion intentional or should that be `&self`? I don't think that methods that take ownership aren't very common in e-g either, at least for larger structs.

Should I rename the method if I change it to return a new rectangle? Because `resize` does sound like it would change the size of an existing rectangle IMO.

Reply:
> But I'm not exactly sure what that is at the moment.

Likewise :thinking: 

> Was the self in you suggestion intentional or should that be &self?

Yes, it could also be `&self`. I thought Clippy might complain but I just tried it and it doesn't so `&self` is a better bet.

> Should I rename the method if I change it to return a new rectangle?

Presumably to `resized`? Yes, probably. It breaks consistency with `translate`, but that should arguably be called `translated`... let's think about that later.

Reply:
> I thought Clippy might complain...

I think Clippy suggests to use `self` instead of `&self` if the size of `Self` is smaller than the pointer size, which is dependent on the target architecture. But a `Rectangle` is larger than a pointer for any common architecture.

> Presumably to `resized`? Yes, probably. It breaks consistency with `translate`, but that should arguably be called `translated`... let's think about that later.

OK, I'll change it to `resized(&self, ...)`.